File: app/src/main/java/myjin/pro/ahoora/myjin/utils/ApiInterface.java
package myjin.pro.ahoora.myjin.utils;

import java.util.List;

import myjin.pro.ahoora.myjin.models.KotlinAboutContactModel;
import myjin.pro.ahoora.myjin.models.KotlinGroupModel;
import myjin.pro.ahoora.myjin.models.KotlinItemModel;
import myjin.pro.ahoora.myjin.models.KotlinMessagesModel;
import myjin.pro.ahoora.myjin.models.KotlinProvCityModel;
import myjin.pro.ahoora.myjin.models.KotlinServicesModel;
import myjin.pro.ahoora.myjin.models.KotlinSignInModel;
import myjin.pro.ahoora.myjin.models.KotlinSlideMainModel;
import myjin.pro.ahoora.myjin.models.KotlinSpecialityModel2;
import myjin.pro.ahoora.myjin.models.SimpleResponseModel;
import myjin.pro.ahoora.myjin.models.TempModel;
import okhttp3.Response;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface ApiInterface {

    @GET("service/getMainGroupList/{provId}/{cityId}")
    Call<List<KotlinGroupModel>> getGroupCount(@Path("provId") int provId, @Path("cityId") int cityId);

    @GET("service/getItems/{gId}/{provId}/{cityId}")
    Call<List<KotlinItemModel>> getItems(@Path("gId") int groupId, @Path("provId") int provId, @Path("cityId") int cityId);

    @GET("service/getSpList/")
    Call<List<KotlinSpecialityModel2>> getSpList();

    @GET("service/search2/{someThing}/{provId}/{cityId}")
    Call<List<KotlinItemModel>> search(@Path("someThing") String searchedText, @Path("provId") int provId, @Path("cityId") int cityId);

    @GET("service/login/{user}/{pass}/{yekta}")
    Call<TempModel> login(@Path("user") String user, @Path("pass") String pass, @Path("yekta") String yekta);

    @GET("service/ac/")
    Call<KotlinAboutContactModel> getAc();

    @GET("service/update/{id}/{lat}/{lng}")
    Call<SimpleResponseModel> updateGeoLocation(@Path("id") int autoId, @Path("lat") double lat, @Path("lng") double lng);

    @GET("service/IsOkServerStatus/")
    Call<TempModel> IsOkServer();

    @GET("service/sliderMain/{slide}")
    Call<List<KotlinSlideMainModel>> sliderMain(@Path("slide") int slide);

    @GET("service/getServicesList/")
    Call<List<KotlinServicesModel>> getServicesList();

    @GET("service/getProvinceAndCitiesList/{cityCount}")
    Call<List<KotlinProvCityModel>> getProvinceAndCitiesList(@Path("cityCount") int cityCount);

    @GET("service/getMessagesList/")
    Call<List<KotlinMessagesModel>> getMessages();

    @FormUrlEncoded
    @POST("sms/src/VerifyLookup.php")
    Call<TempModel>sendSms(@Field("number") String number);



    @FormUrlEncoded
    @POST("service/signin/")
    Call<TempModel>signIn(@Field("p") String number,@Field("f") String fn,@Field("l") String ln,@Field("pr") String pr,@Field("du") String du
            ,@Field("logOut") String logOut,@Field("yekta") String yekta);





    @GET("service/getUserInfo/{number}")
    Call<List<KotlinSignInModel>> getUserInfo(@Path("number") String number);
}

File: app/src/main/java/myjin/pro/ahoora/myjin/utils/ApiInterface.kt
package myjin.pro.ahoora.myjin.utils

import myjin.pro.ahoora.myjin.models.*
import okhttp3.Response
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface ApiInterface {

    @get:GET("service/getSpList/")
    val spList: Call<List<KotlinSpecialityModel2>>

    @get:GET("service/ac/")
    val ac: Call<KotlinAboutContactModel>

    @get:GET("service/getServicesList/")
    val servicesList: Call<List<KotlinServicesModel>>

    @get:GET("service/getMessagesList/")
    val messages: Call<List<KotlinMessagesModel>>

    @GET("service/getMainGroupList/{provId}/{cityId}")
    fun getGroupCount(@Path("provId") provId: Int, @Path("cityId") cityId: Int): Call<List<KotlinGroupModel>>

    @GET("service/getItems/{gId}/{provId}/{cityId}")
    fun getItems(@Path("gId") groupId: Int, @Path("provId") provId: Int, @Path("cityId") cityId: Int): Call<List<KotlinItemModel>>

    @GET("service/search2/{someThing}/{provId}/{cityId}")
    fun search(@Path("someThing") searchedText: String, @Path("provId") provId: Int, @Path("cityId") cityId: Int): Call<List<KotlinItemModel>>

    @GET("service/login/{user}/{pass}/{yekta}")
    fun login(@Path("user") user: String, @Path("pass") pass: String, @Path("yekta") yekta: String): Call<TempModel>

    @GET("service/update/{id}/{lat}/{lng}")
    fun updateGeoLocation(@Path("id") autoId: Int, @Path("lat") lat: Double, @Path("lng") lng: Double): Call<SimpleResponseModel>

    @GET("service/IsOkServerStatus/")
    fun IsOkServer(): Call<TempModel>

    @GET("service/sliderMain/{slide}")
    fun sliderMain(@Path("slide") slide: Int): Call<List<KotlinSlideMainModel>>

    @GET("service/getProvinceAndCitiesList/{cityCount}")
    fun getProvinceAndCitiesList(@Path("cityCount") cityCount: Int): Call<List<KotlinProvCityModel>>

    @FormUrlEncoded
    @POST("sms/src/VerifyLookup.php")
    fun sendSms(@Field("number") number: String): Call<TempModel>


    @FormUrlEncoded
    @POST("service/signin/")
    fun signIn(@Field("p") number: String, @Field("f") fn: String, @Field("l") ln: String, @Field("pr") pr: String, @Field("du") du: String, @Field("logOut") logOut: String, @Field("yekta") yekta: String): Call<TempModel>


    @GET("service/getUserInfo/{number}")
    fun getUserInfo(@Path("number") number: String): Call<List<KotlinSignInModel>>
}
